cli/dcoscli/package/main.py

```diff
@@ -682,10 +682,20 @@ def _bundle_uris(uris_directory, zip_file):
     :rtype: None
     """
 
-    for filename in sorted(os.listdir(uris_directory)):
+    uris = sorted(os.listdir(uris_directory))
+    # these uris will be found through a mustache template from the property
+    # name so make sure each name is unique.
+    uri_names = set("-".join(name.split('.')) for name in uris)
+    if len(uris) != len(uri_names):
+        raise DCOSException(
+            'Error bundling package. Multiple assets map to the same property '
+            'name (periods [.] are replaced with dashes [-]): {}'.format(uris))
+    for filename in uris:
         fullpath = os.path.join(uris_directory, filename)
 
-        zip_file.write(fullpath, arcname='assets/uris/{}'.format(filename))
+        # s/./- so can be rendered by mustache template.
+        zip_file.write(fullpath, arcname='assets/uris/{}'.format(
+            "-".join(filename.split("."))))
```

Comment:
Wait. Ignore my comment above. We shouldn't be renaming the file. We should only be check that renaming the property doesn't collide. The reason why we perform the check above is so that we guarantee that two files don't have the same property name.

We do this because mustache resolve properties with `.` differently. The reason why we can't change the filename is because the code executing in the Mesos task expects the files to have the same name.